packages/amplify-cli/index.js

```diff
@@ -0,0 +1,98 @@
+const { existsSync, mkdirSync, createWriteStream } = require('fs');
+const { join } = require('path');
+const { spawnSync } = require('child_process');
+const util = require('util');
+const stream = require('stream');
+const pipeline = util.promisify(stream.pipeline);
+const os = require('os');
+const axios = require('axios');
+const tar = require('tar');
+const rimraf = require('rimraf');
+
+const error = msg => {
+  console.error(msg);
+  process.exit(1);
+};
+
+class Binary {
+  constructor(name, url) {
+    let errors = [];
+    if (typeof url !== 'string') {
+      errors.push('url must be a string');
+    } else {
+      try {
+        new URL(url);
+      } catch (e) {
+        errors.push(e);
+      }
+    }
+    if (name && typeof name !== 'string') {
+      errors.push('name must be a string');
+    }
+
+    if (!name) {
+      errors.push('You must specify the name of your binary');
+    }
+    if (errors.length > 0) {
+      let errorMsg = 'One or more of the parameters you passed to the Binary constructor are invalid:\n';
+      errors.forEach(error => {
+        errorMsg += error;
+      });
+      errorMsg += '\n\nCorrect usage: new Binary("my-binary", "https://example.com/binary/download.tar.gz")';
+      error(errorMsg);
+    }
+    this.url = url;
+    this.name = name;
+    this.installDirectory = join(os.homedir(), '.amplify', 'bin');
```

Comment:
this is where the binary will live. originally I wanted to include in the npm bin dir, but that does not work properly for sudo installs.